docs/TerraToolkitUpgradeGuide-v5.0.0.md

```diff
@@ -0,0 +1,116 @@
+# Terra Toolkit Upgrade Guide v5.0.0
+This document will provide information on upgrading from terra-toolkit 4.x to 5.0.0.
+
+## Terra Scripts
+### Serve
+Serve is now a thin abstraction on webpack dev server and the commandline api is now identical. With this addition it now means that you have control over the dev server through options specified in your webpack config as well as through the cli.
+
+Why use serve instead of webpack-dev-server directly? Having the serve abstraction provides a hook for us to change the servers implementation in case webpack-dev-server nolonger meets our needs.
+
+Our default webpack config has been updated with the following defaults:
+
+```javascript
+module.exports = {
+  devServer: {
+    // We set the host this way to allow it to be served from docker containers.
+    host: '0.0.0.0',
+    // These options cut down on noise in the webpack build output.
+    stats: {
+      colors: true,
+      children: false,
+    },
+  },
+};
+```
+
+Serve is nolonger available as a javascript function. If this is needed use webpack-dev-server directly.
+
+### Serve Static
+Webpack-dev-server now supports IE 10+, because of this change we have removed the ability for serve-static to run webpack to create your site. Serve static now will simply host static site content.
+
+Serve static will now return 404.html (if it exists) if an html page is not found.
+
+These api options have been removed from both the cli and javascript:
+* config
+* production
+* disk
+
+If you want to serve a non hot-reloading site without pre-building your site, use tt-serve-static with the following options specified in your webpack config.
+
+```javascript
+module.exports = {
+  devServer: {
+    // Disable hot reloading and javascript injection to watch for changes.
+    hot: false,
+    inline: false,
+    },
```

Comment:
I think it would be helpful to provide this config default. We are adding this in the ServeStatic service automatically when compiling assets for tests so we need a quick way for people to reproduce what wdio is using. I think this would be better if custom config was not needed to achieve that.

Since we are wrapping the webpack-dev-server CLI we can add a webpack env var that we can check in the webpack config that will allow people to disabled via the command line if hot loading is undesirable. We already have this option available with disabling aggregate translations or passing locale options through webpack.

This would allow users to check their static production assets like
```bash
npm run tt-serve --env.disableHotReloading -p
```

webpack.config.sj changes:

```diff
const webpackConfig = (options, env, argv) => {
-  const { rootPath, resolveModules, themeFile } = options;
+  const { rootPath, resolveModules, themeFile, staticOptions } = options;
   ...
   const config = {
    ....
    devServer: {
+    ...staticOptions,
      host: '0.0.0.0',
      stats: {
        colors: true,
        children: false,
      },
    },
   };

  return config;
};

const defaultWebpackConfig = (env = {}, argv = {}) => {
-  const { disableAggregateTranslations } = env;
+  const { disableAggregateTranslations, disableHotReloading } = env;

....

+  let staticOptions = {};
+  if (!disableHotReloading) {
+    staticOptions = {
+        hot: false,
+      inline: false,
+    };
+ }

  const themeFile = ThemeAggregator.aggregate();

- const options = { rootPath, resolveModules, themeFile };
+ const options = { rootPath, resolveModules, themeFile, staticOptions };

  return webpackConfig(options, env, argv);
};

module.exports = defaultWebpackConfig;
```

Reply:
done